cql3/statements/describe_statement.cc

```diff
@@ -282,7 +282,9 @@ future<std::vector<description>> table(const data_dictionary::database& db, cons
 }
 
 future<std::vector<description>> tables(const data_dictionary::database& db, const lw_shared_ptr<keyspace_metadata>& ks, std::optional<bool> with_internals = std::nullopt) {
```

Comment:
Does this mean that we hide CDC tables only when listing the whole schema or a keyspace? What about describing the CDC table itself?

I'm not entirely sure how we should handle it.

- On one hand, if we allow describing cdc tables then this sounds a bit inconsistent to me - the table exists when you ask for it explicitly but does not when listing.
- On the other hand, if we don't allow describing cdc tables then some information becomes inaccessible (e.g. various parameters like gc_grace_seconds of the table).

My questions that should help with making the decision are:

- The situation is completely analogous to materialized views backing secondary index - how to we handle it there?
- Is it possible to run ALTER on a cdc log table? If so, then allowing DESCRIBE SCHEMA on it has some value. However, this would mean that restoring schema from DESCRIBE SCHEMA after the changes you implemented here is broken because restoring schema would lose the changes that the user introduced. In that case I think we should consider emitting an ALTER statement ourselves in DESCRIBE SCHEMA etc.

Reply:
The question about ALTER is interesting, and also applies to materialized views :-( We also allow "ALTER MATERIALIZED VIEW", for example `ALTER MATERIALIZED VIEW ks.v WITH synchronous_updates = true;`. Materialized views are normally visible with DESC TABLES but materialized views that back an *index* are not, and the information that we changed something on that view will not be visible. The MV issue is related to https://github.com/scylladb/scylladb/issues/16454 so I'll comment about it there.